### PR TITLE
[PAR-4327] admin cart balancing setting toggle

### PR DESCRIPTION
**Pull request by aarondeane-extend**
# Description
- Add toggle to admin to enable/disable cart balancing
- Add backend configuration to save at a `default` level

## Ticket
https://helloextend.atlassian.net/browse/PAR-4327

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have deployed and validated my code on a sandbox
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: etc/adminhtml/system.xml [hunk 1]
```diff
@@ -40,20 +40,26 @@
...
+            <group id="product_protection" translate="label" type="text" sortOrder="20" showInDefault="1" showInWebsite="1" showInStore="1">
                 <label>Product Protection</label>
                 <field id="enable" translate="label" type="select" sortOrder="10" showInDefault="1" canRestore="1">
                     <label>Enable Product Protection (V2)</label>
                     <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
                     <frontend_model>Extend\Integration\Model\Config\Frontend\EnableProductProtection</frontend_model>
                     <backend_model>Extend\Integration\Model\Config\Backend\EnableProductProtection</backend_model>
                 </field>
-                <field id="recreate_pp_product" type="button" sortOrder="20" showInDefault="1">
+                <field id="recreate_pp_product" type="button" sortOrder="20" showInDefault="1" canRestore="1">
```
Comment (milesmcleod) [inline etc/adminhtml/system.xml:51]: i dunno what `canRestore` does for a `button` field - probably not an issue, but maybe not necessary?

File: etc/adminhtml/system.xml [hunk 2]
```diff
@@ -40,20 +40,26 @@
...
                     <button_label>Recreate Product Protection Product</button_label>
                     <button_url>extend_integration/productprotection/create</button_url>
                     <frontend_model>Extend\Integration\Model\Config\Frontend\RecreatePPProduct</frontend_model>
                 </field>
-                <group id="offer_display_settings" translate="label" type="text" sortOrder="20" showInDefault="1">
+                <field id="enable_cart_balancing" translate="label" type="select" sortOrder="30" showInDefault="1" showInWebsite="1" showInStore="1" canRestore="1">
```
Comment (milesmcleod) [inline etc/adminhtml/system.xml:56]: probably want the `sortOrder` of this field to be `20` if we want it pinned below the `Recreate` button

**Review by milesmcleod — CHANGES_REQUESTED** [review]

File: etc/adminhtml/system.xml (hunk 2 above)
Comment (aarondeane-extend) [reply etc/adminhtml/system.xml:56]: the recreate button in this is set to `20` though so I had set it to `30`, does it show up weird for you?

Comment (milesmcleod) [reply etc/adminhtml/system.xml:56]: whoops, didn't notice that. in that case we should leave this at `30` and probably bump the `offer_display_settings` group below up to 40, right? that'd be sequential

Comment (aarondeane-extend) [reply etc/adminhtml/system.xml:56]: good call

File: etc/adminhtml/system.xml (hunk 1 above)
Comment (aarondeane-extend) [reply etc/adminhtml/system.xml:51]: from what I've read this just allows you to restore it to the default value from admin without having to manually enter it. Pretty much just copied the other implementation of this field type

**Review by milesmcleod — APPROVED** [review]

File: Model/Config/Backend/Enable.php [hunk 3]
```diff
@@ -52,6 +52,7 @@ public function afterSave()
         if ($isV2Enabled === 0) {
             $this->writer->save(\Extend\Integration\Service\Extend::ENABLE_PRODUCT_PROTECTION, 0);
             $this->writer->save(\Extend\Integration\Service\Extend::ENABLE_SHIPPING_PROTECTION, 0);
+            $this->writer->save(\Extend\Integration\Service\Extend::ENABLE_CART_BALANCING, 0);
```
Comment (adityaagashe1) [inline Model/Config/Backend/Enable.php:55]: By Default, we want to turn off the cartBalancing?

**Review by adityaagashe1 — APPROVED** [review]

**Pull request merged**